src/components/Widget/inputs.tsx

```diff
@@ -0,0 +1,66 @@
+import { Currency, Field, SwapController, SwapEventHandlers, TradeType } from '@uniswap/widgets'
+import CurrencySearchModal from 'components/SearchModal/CurrencySearchModal'
+import { useCallback, useMemo, useState } from 'react'
+
+// Integrates the Widget's controlled value, using the app's CurrencySearchModal for token selection.
+export function useSyncWidgetInputs(defaultToken: Currency) {
+  const [isExactInput, setIsExactInput] = useState(false)
+  const [amount, setAmount] = useState<string>()
+  const onAmountChange = useCallback((field: Field, amount: string) => {
+    setIsExactInput(field === Field.INPUT)
+    setAmount(amount)
+  }, [])
+  const [tokens, setTokens] = useState<{ [Field.INPUT]?: Currency; [Field.OUTPUT]?: Currency }>({
+    [Field.OUTPUT]: defaultToken,
+  })
+  const onSwitchTokens = useCallback(() => {
+    setIsExactInput((isExactInput) => !isExactInput)
+    setTokens((tokens) => ({
+      [Field.INPUT]: tokens[Field.OUTPUT],
+      [Field.OUTPUT]: tokens[Field.INPUT],
+    }))
+  }, [])
+  const [selectingField, setSelectingField] = useState<Field>()
+  const otherField = useMemo(() => (selectingField === Field.INPUT ? Field.OUTPUT : Field.INPUT), [selectingField])
+  const [selectingToken, otherToken] = useMemo(() => {
+    if (selectingField === undefined) return [undefined, undefined]
+    return [tokens[selectingField], tokens[otherField]]
+  }, [otherField, selectingField, tokens])
+  const onTokenSelectorClick = useCallback((field: Field) => {
+    setSelectingField(field)
+    return false
+  }, [])
+  const onTokenSelect = useCallback(
+    (token: Currency) => {
+      if (selectingField === undefined) return
+      setIsExactInput(selectingField === Field.INPUT)
+      setTokens(() => {
+        return {
+          [otherField]: token === otherToken ? selectingToken : otherToken,
+          [selectingField]: token,
+        }
+      })
+    },
+    [otherField, otherToken, selectingField, selectingToken]
+  )
+  const value: SwapController = useMemo(
+    () => ({ type: isExactInput ? TradeType.EXACT_INPUT : TradeType.EXACT_OUTPUT, amount, ...tokens }),
```

Comment:
same issue as above, feel like this could just be `type` from a replacement for `isExactInput` that uses the `TradeType` itself instead of a bool